python/cudf/cudf/core/dataframe.py

```diff
@@ -2359,7 +2359,7 @@ def to_dict(
         You can also specify the mapping type.
 
         >>> from collections import OrderedDict, defaultdict
-        >>> df.to_dict(into=OrderedDict)
+        >>> df.to_dict(into=OrderedDict)  # doctest: +SKIP
```

Comment:
Starting with Python 3.12, the `repr()` representation for an `OrderedDict` has changed:

* https://github.com/python/cpython/issues/101446
* https://docs.python.org/3/whatsnew/changelog.html

*(thanks @bdice for pointing me to that changelog item!)*

That's causing doctest failures like this:

```text
E       AssertionError: 1 of 13 doctests failed for DataFrame.to_dict:
E         **********************************************************************
E         File "/opt/conda/envs/test/lib/python3.12/site-packages/cudf/core/dataframe.py", line 2362, in DataFrame.to_dict
E         Failed example:
E             df.to_dict(into=OrderedDict)
E         Expected:
E             OrderedDict([('col1', OrderedDict([('row1', 1), ('row2', 2)])),
E                          ('col2', OrderedDict([('row1', 0.5), ('row2', 0.75)]))])
E         Got:
E             OrderedDict({'col1': OrderedDict({'row1': 1, 'row2': 2}), 'col2': OrderedDict({'row1': 0.5, 'row2': 0.75})})
E         **********************************************************************
E         1 items had failures:
E            1 of  13 in DataFrame.to_dict
E         ***Test Failed*** 1 failures.
E         
E       assert not 1
E        +  where 1 = TestResults(failed=1, attempted=13).failed

/__w/cudf/cudf/python/cudf/cudf/tests/test_doctests.py:125: AssertionError
```

This proposes just skipping those tests, here and in the docs for `Series.to_dict()`.